pymc/logprob/transforms.py

```diff
@@ -523,8 +539,54 @@ def measurable_sub_to_neg(fgraph, node):
     return [pt.add(minuend, pt.neg(subtrahend))]
 
 
+@node_rewriter([log1p, softplus, log1mexp, log2, log10])
+def measurable_special_log_to_log(fgraph, node):
+    """Convert log1p, log1mexp, softplus of `MeasurableVariable`s to log form."""
+    [inp] = node.inputs
+
+    if isinstance(node.op.scalar_op, Log1p):
+        return [pt.log(1 + inp)]
+    if isinstance(node.op.scalar_op, Softplus):
+        return [pt.log(1 + pt.exp(inp))]
+    if isinstance(node.op.scalar_op, Log1mexp):
+        return [pt.log(1 - pt.exp(pt.neg(inp)))]
+    if isinstance(node.op.scalar_op, Log2):
+        return [pt.log(inp) / pt.log(2)]
+    if isinstance(node.op.scalar_op, Log10):
+        return [pt.log(inp) / pt.log(10)]
+
+
+@node_rewriter([exp2, expm1, sigmoid])
+def measurable_special_exp_to_exp(fgraph, node):
+    """Convert log1p, log1mexp, softplus of `MeasurableVariable`s to log form."""
+    [inp] = node.inputs
+
+    if isinstance(node.op.scalar_op, Exp2):
+        return [pt.power(2, inp)]
```

Comment:
I don't think we support this one, only powers with fixed exponent.

Reply:
You can convert to `exp(x*ln(2))` instead, which PyMC will know how to handle

Reply:
We should actually use that for any `power(const, x)` which we currently don't support.

Reply:
Ah yeah I had a feeling that one may be an issue, I'll make the change. I'll add a new function to generalise this power(const, x) -> exp(log(const) * x) functionality.

Reply:
So I've just pushed this new functionality. Do we care that it won't work const <= 0?

Reply:
Let's open a separate issue / PR for this. I thought it over a bit and I think we can definitely support a couple of cases.

```
power(const, x), for any const > 0 and any x
power(neg_const, x), for any neg_const < 0, and discrete x  (we can play with `log(abs(neg_const))` and x's parity)
```
The second requires us to implement transforms for discrete variables, which would probably need #6360 first, so we can focus on the first, which is also probably more useful.

We just have to make sure not to rewrite stuff like `power(x, const)` accidentally as those are implemented via our `PowerTransform`. This is done by checking which inputs have a path to unvalued random variables